libvast/src/detail/mmapbuf.cpp

```diff
@@ -105,6 +105,15 @@ bool mmapbuf::resize(size_t new_size) {
   // Resize the underlying file, if available.
   if (fd_ != -1 && ftruncate(fd_, new_size) < 0)
     return false;
+#if __linux__
```

Comment:
We wrap the various platform-dependent defines: `#ifdef VAST_LINUX` (make sure `config.hpp` is included).